### ISSUE TITLE
Support dependency injection in lambda parameters

### ISSUE THREAD

**Issue by RobinVds** [awslabs/aws-lambda-powertools-python#924]
**Is your feature request related to a problem? Please describe.**
I'm injecting my dependencies in my lambda functions to be able to easily test my functions.

```
@logger.inject_lambda_context(log_event=True)
def lambda_handler(event,
                   context,
                   s3_client=s3,
                   sqs_queue=sqs_queue):
```

**Describe the solution you'd like**
Would it be possible to just allow more parameters in the decorator function than only event and context? Just pass all the parameters that are included in the lambda function call.

**Describe alternatives you've considered**
Not relying on dependency injection and testing one level lower.

**Additional context**
Not sure if this way of testing lambdas is a best practice but it makes it a lot easier that's for sure.

**Comment by heitorlessa** [comment]
Hey @RobinVds i merged that one today - it should be in the next release before the end of the year.

That said, there are better ways to test it - I often use dependency injection with ports and adapters so the handler becomes just an entry point.

I can post an example later and I've been meaning to update our quick start template to demonstrate how easy it becomes for unit and Integ test

**Comment by RobinVds** [comment]
> Hey @RobinVds i merged that one today - it should be in the next release before the end of the year.
> 
> That said, there are better ways to test it - I often use dependency injection with ports and adapters so the handler becomes just an entry point.
> 
> I can post an example later and I've been meaning to update our quick start template to demonstrate how easy it becomes for unit and Integ test

Sounds interesting, looking forward to seeing how you implement that! :+1:

**Comment by RobinVds** [comment]
@heitorlessa Would it be possible to post the example? Thanks!

**Comment by heitorlessa** [comment]
See if you have access to this: https://github.com/heitorlessa/aws-serverless-airline-booking/blob/refactor%2Floyalty-runtime/src/backend/loyalty/src/loyalty/aggregate/app.py#L17

* Storage Port: https://github.com/heitorlessa/aws-serverless-airline-booking/blob/refactor%2Floyalty-runtime/src/backend/loyalty/src/loyalty/shared/storage.py#L41
* Adapter for unit tests: https://github.com/heitorlessa/aws-serverless-airline-booking/blob/refactor%2Floyalty-runtime/src/backend/loyalty/src/loyalty/shared/storage.py#L55
* DynamoDB Adapter: https://github.com/heitorlessa/aws-serverless-airline-booking/blob/refactor%2Floyalty-runtime/src/backend/loyalty/src/loyalty/shared/storage.py#L88
* Unit tests using FakeStorage adapter: https://github.com/heitorlessa/aws-serverless-airline-booking/blob/refactor%2Floyalty-runtime/src/backend/loyalty/tests/unit/test_aggregate.py


I can provide more details tomorrow as I've been writing a doc for the past few days.

**Comment by RobinVds** [comment]
Thanks for the info! If you could, please ping me when you are finished with the doc :)

**Comment by heitorlessa** [comment]
hey @RobinVds this is available now in the 1.24 release (since Dec 31st tbh ;)). I've published the talk as a PDF covering that code snippet above - slide 46 and onward for the Ports & Adapters alone: https://www.slideshare.net/HeitorLessa1/aws-community-day-ireland-refactoring-a-serverless-app

This feature request is also now available in the 1.24 release - thanks a lot for raising that with us!
